### PR TITLE
Add support for automation description

### PR DESCRIPTION
**Pull request by emontnemery**
When using the automation editor, any manually added comments are wiped.
Allow adding `description` string to the automation instead, with the same purpose as a comment.

Depends on backend PR: https://github.com/home-assistant/home-assistant/pull/26662

Screenshots:
image

image

### CODE REVIEW

**Comment by SeanPM5** [comment]
I really love this idea. Both for this particular use case of remembering logic in your personal automations, but also for the future if we get automation sharing, this would be a great way to describe your shared automation to the community.

I'm just not sure that everyone will know what "Annotation" means? I know it's the correct word but I think that "Description", "Note", or "Comment" might be more easily understandable to a wider group of people IMO.

Also I think that the label and/or placeholder text should make clear that it's optional too.

**Comment by emontnemery** [comment]
@SeanPM5 OK, have a look at the updated screenshots, maybe better now.
Do you think it should be mentioned in the placeholder that the comment can be multple lines?

**Comment by balloob** [comment]
I like the idea, but let's call it description.

**Comment by emontnemery** [comment]
@balloob Fixed!

File: src/translations/en.json [hunk 1]
```diff
@@ -699,13 +699,17 @@
             "add_automation": "Add automation"
           },
           "editor": {
-            "introduction": "Use automations to bring your home alive",
+            "introduction": "Use automations to bring your home alive. Automations rules can be complicated, optionally add a description to document what it does and how.",
```
Comment (balloob) [inline src/translations/en.json]: Let's drop this. It's the least important feature on this page and so should not require an individual call out.

Comment (emontnemery) [reply src/translations/en.json]: OK, removed!

**Comment by balloob** [comment]
We might have to do one more frontend release for the final release. As this PR relies on backend code that is not in tomorrows release , we should not merge this yet.

**Review by balloob — APPROVED** [review]

**Pull request merged**